packages/@groupby/elements-cache-driver-plugin/src/cache-driver-plugin.ts

```diff
@@ -0,0 +1,61 @@
+// eslint-disable-next-line import/no-extraneous-dependencies, import/no-unresolved
+import { Plugin, PluginMetadata, PluginRegistry } from '@groupby/elements-core';
+import { CACHE_REQUEST, CacheRequestPayload, CacheResponsePayload } from '@groupby/elements-events';
+
+/**
+ * The GroupBy Elements cache driver plugin.
+ * This plugin exposes an event-based interface to the GB Elements cache.
+ */
+export default class CacheDriverPlugin implements Plugin {
+  get metadata(): PluginMetadata {
+    return {
+      name: 'cache_driver',
+      depends: ['cache', 'dom_events'],
+    };
+  }
+
+  /**
+   * A reference to the registry of plugins for internal use.
+   */
+  core: PluginRegistry;
+
+  /**
+   * Constructs a new instance of this plugin and binds the necessary
+   * callbacks.
+   */
+  constructor() {
+    this.handleRequest = this.handleRequest.bind(this);
+  }
+
+  /**
+   * Saves the plugin registry for later use. This plugin does not
+   * expose a value.
+   *
+   * @param plugins the plugin registry to use.
+   */
+  register(plugins: PluginRegistry): void {
+    this.core = plugins;
+  }
+
+  /**
+   * Registers a number of event listeners.
+   * The following events are listened for:
+   *
+   * - [[CACHE_REQUEST]]
+   */
+  ready(): void {
+    this.core.dom_events.registerListener(CACHE_REQUEST, this.handleRequest);
+  }
+
+  /**
+   * Dispatches the requested data from the cache.
+   *
+   * @param req The cache request details.
+   */
+  handleRequest(req: CustomEvent<CacheRequestPayload>): void {
```

Comment:
`CACHE_ERROR` was for the case where there is an error between this plugin requesting data from the cache and the cache returning the requested data. I guess we can't guarantee that the cache is what it claims to be, so perhaps it would be good to wrap it in a `try..catch` in case there are any errors. After a discussion with @DanielMoniz-GBI, whether or not we decide to do it, we will push it off to a later date and this PR can go through without it.